openvdb_houdini/SOP_OpenVDB_Activate.cc

```diff
@@ -0,0 +1,604 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+//
+/// @file SOP_OpenVDB_Activate.cc
+///
+/// @author FX R&D OpenVDB team
+///
+/// @brief Activate VDBs according to various rules
+
+#if 1
+  #include <houdini_utils/ParmFactory.h>
+  #include <openvdb_houdini/Utils.h>
+  #include <openvdb_houdini/SOP_NodeVDB.h>
+#else
+  #include "Utils.h"
+  #include "ParmFactory.h"
+  #include "SOP_NodeVDB.h"
+#endif
+
+#include <GU/GU_PrimVDB.h>
+#include <OP/OP_Node.h>
+#include <OP/OP_Operator.h>
+#include <OP/OP_OperatorTable.h>
+#include <PRM/PRM_Parm.h>
+
+#include <openvdb/openvdb.h>
+#include <openvdb/Types.h>
+#include <openvdb/tools/Morphology.h>
+#include <openvdb/tools/Prune.h>
+
+namespace hvdb = openvdb_houdini;
+namespace hutil = houdini_utils;
+
+using namespace openvdb_houdini;
+
+enum REGIONTYPE_NAMES
+{
+    REGIONTYPE_POSITION,
+    REGIONTYPE_VOXEL,
+    REGIONTYPE_EXPAND,
+    REGIONTYPE_REFERENCE,
+    REGIONTYPE_DEACTIVATE,
+    REGIONTYPE_FILL
```

Comment:
It is intended to activate all tiles and set them all to the negative background value.  It is intended to not densify in the process.  Whether I succeeded in that is up for debate.

I think the off iterator will return tiles as a single bbox result.  I then trigger the fill() function that should preserve the tile sparsity and activate & set it's value.

The primary goal for me is to get a topology grid that has the interior of an object fully activated.  But I think there should be little harm/cost in also ensuring that the interior voxel values are "correct" rather than acquiring ghost values.

Reply:
Right, that makes sense. Maybe we would need to rethink the tab name - VDB Rebuild SDF and others have a "fill interior" toggle which could be easily confused from this one (that's what I wrongly assumed it was on first seeing it).